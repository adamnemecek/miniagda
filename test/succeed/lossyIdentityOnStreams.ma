
sized codata Stream (+ A : Set) : Size -> Set {
  cons : (i : Size) -> A -> Stream A i -> Stream A ($ i)
}

cofun sid : (A : Set) -> (i : Size) -> Stream A ($ i) -> Stream A i
{
  sid A ($ i) (cons .A .($ i) x xs) = cons A _ x (sid A i xs)
}
