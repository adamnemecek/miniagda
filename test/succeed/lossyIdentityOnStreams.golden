--- opening "lossyIdentityOnStreams.ma" ---
--- scope checking ---
--- type checking ---
type  Stream : ++(A : Set) -> - Size -> Set
term  Stream.cons : .[A : Set] -> .[i : Size] -> ^(y1 : A) -> ^(y2 : Stream A i) -> < Stream.cons i y1 y2 : Stream A $i >
term  sid : .[A : Set] -> .[i : Size] -> Stream A $i -> Stream A i
{ sid [A] $[i < #] (Stream.cons [.$i] x xs) = Stream.cons [i] x (sid [A] [i] xs)
}
--- evaluating ---
--- closing "lossyIdentityOnStreams.ma" ---
